src/ESPEasyRules.ino

```diff
@@ -304,28 +304,84 @@ void replace_EventValueN_Argv(String& line, const String& argString, unsigned in
   }
 }
 
-void substitute_eventvalue(String& line, const String& event) {
-  if (line.indexOf(F("%eventvalue")) == -1) {
-    return; // Nothing to replace.
+void process_substring(String& line) {
+  // Syntax like 12345[substring:8:12:ANOTHER HELLO WORLD]67890
+  int startIndex;
+  while ((startIndex = line.indexOf(F("[substring:"))) != -1) {  
+    String substCmd = line.substring(startIndex, line.indexOf("]", startIndex) + 1);
+    int idxStart = substCmd.indexOf(":") + 1;
+    int idxEnd = substCmd.indexOf(":", idxStart);
+    String substStart = substCmd.substring(idxStart, idxEnd);
+    idxStart = idxEnd + 1;
+    idxEnd = substCmd.indexOf(":", idxStart);
+    String substEnd = substCmd.substring(idxStart , idxEnd);
+    idxStart = idxEnd + 1;
+    idxEnd = substCmd.indexOf("]", idxStart);
+    String stringValue = substCmd.substring(idxStart, idxEnd);
+  
+    line.replace(substCmd, stringValue.substring(substStart.toInt(), substEnd.toInt()));
```

Comment:
You don't check here if the `substStart` or `substEnd` is actually an int.
See: https://github.com/letscontrolit/ESPEasy/blob/0c6e6ca915e5440de54213246dc8c1a8e6ca9cce/src/Misc.ino#L1320-L1327

Reply:
I just added the checks in a more recent commit: 
https://github.com/crathje/ESPEasy/commit/d238593ac0b039cdc10a0b920de9a3fb20f128e7